File: src/ympa_csharp_functions_server/Functions/CampaignsApi.cs
using System.IO;
using System.Net;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
using Microsoft.Extensions.Logging;
using Microsoft.OpenApi.Models;
using Newtonsoft.Json;
using Org.OpenAPITools.Models;

namespace ympa_csharp_functions_server.Functions
{ 
    public partial class CampaignsApi
    { 
        [FunctionName("CampaignsApi_GetCampaign")]
        public async Task<ActionResult<GetCampaignResponse>> _GetCampaign([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("GetCampaign");
            return method != null
                ? (await ((Task<GetCampaignResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("CampaignsApi_GetCampaignLogins")]
        public async Task<ActionResult<GetCampaignLoginsResponse>> _GetCampaignLogins([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}/logins")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("GetCampaignLogins");
            return method != null
                ? (await ((Task<GetCampaignLoginsResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("CampaignsApi_GetCampaignRegion")]
        public async Task<ActionResult<GetCampaignRegionResponse>> _GetCampaignRegion([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}/region")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("GetCampaignRegion");
            return method != null
                ? (await ((Task<GetCampaignRegionResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("CampaignsApi_GetCampaignSettings")]
        public async Task<ActionResult<GetCampaignSettingsResponse>> _GetCampaignSettings([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/{campaignId}/settings")]HttpRequest req, ExecutionContext context, long campaignId)
        {
            var method = this.GetType().GetMethod("GetCampaignSettings");
            return method != null
                ? (await ((Task<GetCampaignSettingsResponse>)method.Invoke(this, new object[] { req, context, campaignId })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("CampaignsApi_GetCampaigns")]
        public async Task<ActionResult<GetCampaignsResponse>> _GetCampaigns([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns")]HttpRequest req, ExecutionContext context)
        {
            var method = this.GetType().GetMethod("GetCampaigns");
            return method != null
                ? (await ((Task<GetCampaignsResponse>)method.Invoke(this, new object[] { req, context })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }

        [FunctionName("CampaignsApi_GetCampaignsByLogin")]
        public async Task<ActionResult<GetCampaignsResponse>> _GetCampaignsByLogin([HttpTrigger(AuthorizationLevel.Anonymous, "Get", Route = "campaigns/by_login/{login}")]HttpRequest req, ExecutionContext context, string login)
        {
            var method = this.GetType().GetMethod("GetCampaignsByLogin");
            return method != null
                ? (await ((Task<GetCampaignsResponse>)method.Invoke(this, new object[] { req, context, login })).ConfigureAwait(false))
                : new StatusCodeResult((int)HttpStatusCode.NotImplemented);
        }
    }
}

File: src/main/java/org/openapitools/server/api/verticle/CampaignsApi.java
package org.openapitools.server.api.verticle;

import org.openapitools.server.api.model.ApiClientDataErrorResponse;
import org.openapitools.server.api.model.ApiForbiddenErrorResponse;
import org.openapitools.server.api.model.ApiLimitErrorResponse;
import org.openapitools.server.api.model.ApiNotFoundErrorResponse;
import org.openapitools.server.api.model.ApiServerErrorResponse;
import org.openapitools.server.api.model.ApiUnauthorizedErrorResponse;
import org.openapitools.server.api.model.GetCampaignLoginsResponse;
import org.openapitools.server.api.model.GetCampaignRegionResponse;
import org.openapitools.server.api.model.GetCampaignResponse;
import org.openapitools.server.api.model.GetCampaignSettingsResponse;
import org.openapitools.server.api.model.GetCampaignsResponse;
import org.openapitools.server.api.MainApiException;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

import java.util.List;
import java.util.Map;

public interface CampaignsApi  {
    //getCampaign
    void getCampaign(Long campaignId, Handler<AsyncResult<GetCampaignResponse>> handler);

    //getCampaignLogins
    void getCampaignLogins(Long campaignId, Handler<AsyncResult<GetCampaignLoginsResponse>> handler);

    //getCampaignRegion
    void getCampaignRegion(Long campaignId, Handler<AsyncResult<GetCampaignRegionResponse>> handler);

    //getCampaignSettings
    void getCampaignSettings(Long campaignId, Handler<AsyncResult<GetCampaignSettingsResponse>> handler);

    //getCampaigns
    void getCampaigns(Integer page, Integer pageSize, Handler<AsyncResult<GetCampaignsResponse>> handler);

    //getCampaignsByLogin
    void getCampaignsByLogin(String login, Integer page, Integer pageSize, Handler<AsyncResult<GetCampaignsResponse>> handler);

}
